itou/www/apply/tests/tests_list.py

```diff
@@ -4,209 +4,344 @@
-)
-from itou.job_applications.models import JobApplicationWorkflow, JobApplication
-from itou.users.factories import DEFAULT_PASSWORD, PrescriberFactory
+from itou.job_applications.models import JobApplicationWorkflow
 from itou.job_applications.factories import JobApplicationSentByPrescriberFactory
+from itou.prescribers.factories import (
+    PrescriberOrganizationWithMembershipFactory,
+    AuthorizedPrescriberOrganizationWithMembershipFactory,
+)
+from itou.siaes.factories import SiaeWithMembershipAndJobsFactory
 from itou.utils.widgets import DatePickerField
+from itou.users.factories import DEFAULT_PASSWORD, PrescriberFactory
 
 
-class ProcessListJobSeekerAndSIAETest(TestCase):
+class ProcessListTest(TestCase):
     def setUp(self):
-        # Create 1 JobApplication per available state and attach it to the same SIAE.
-        siae = SiaeWithMembershipFactory()
+        """
+        Create three organizations with two members each:
+        - pole_emploi: job seekers agency.
+        - l_envol: an emergency center for homeless people.
+        - hit_pit: a boxing gym looking for boxers.
+
+        Pole Emploi prescribers:
+        - Thibault
+        - laurie
+
+        L'envol prescribers:
+        - Audrey
+        - Manu
+
+        Hit Pit staff:
+        - Eddie
+        """
+
+        # Pole Emploi
+        pole_emploi = AuthorizedPrescriberOrganizationWithMembershipFactory(
+            name="Pôle Emploi"
+        )
+        laurie_pe = PrescriberFactory(first_name="Laurie")
+        thibault_pe = pole_emploi.members.first()
+        thibault_pe.save(update_fields={"first_name": "Thibault"})
+        pole_emploi.members.add(laurie_pe)
+
+        # L'Envol
+        l_envol = PrescriberOrganizationWithMembershipFactory(name="L'Envol")
+        manu_envol = PrescriberFactory(first_name="Manu")
+        audrey_envol = l_envol.members.first()
+        audrey_envol.save(update_fields={"first_name": "Audrey"})
+        l_envol.members.add(manu_envol)
+
+        # Hit Pit
+        hit_pit = SiaeWithMembershipAndJobsFactory(name="Hit Pit")
+        eddie_hit_pit = hit_pit.members.first()
+        eddie_hit_pit.first_name = "Eddie"
+        eddie_hit_pit.save()
+
+        # Now send applications
         for i, state in enumerate(JobApplicationWorkflow.states):
             creation_date = timezone.now() - timezone.timedelta(days=i)
-            job_application = JobApplicationFactory(
-                to_siae=siae, state=state, created_at=creation_date
+            job_application = JobApplicationSentByPrescriberFactory(
+                to_siae=hit_pit,
+                state=state,
+                created_at=creation_date,
+                sender=thibault_pe,
+                sender_prescriber_organization=pole_emploi,
             )
 
-        # SIAE view
-        self.siae_user = job_application.to_siae.members.first()
-        self.siae_base_url = reverse("apply:list_for_siae")
+        maggie = job_application.job_seeker
```

Comment:
Ça fait un peu bizarre de se référer à `job_application` qui a été déclarée dans un `for`. Qu'est-ce que tu as voulu dire dans ton test exactement ?

Reply:
Je suis d'accord avec toi, c'est vrai que ça fait bizarre. Mais ça correspond à ce que je voulais faire :  récupérer le demandeur d'emploi de la dernière candidature envoyée par `thibault_pe`. J'aurais pu faire une `query` mais vu que l'objet est créé juste avant, ça évite de faire une nouvelle requête dans la base.